code/game/objects/items/weapons/inducer.dm

```diff
@@ -0,0 +1,190 @@
+								 "<span class='notice'> You recharged the [O]!</span>")
+			return
+	..()
+
+/obj/item/weapon/inducer/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W,/obj/item/weapon/screwdriver))
+		if(!opened)
+			to_chat(user, "<span class='notice'>You unscrew the battery compartment.</span>")
+			opened = TRUE
+			update_icon()
+			return
+		else
+			to_chat(user, "<span class='notice'>You close the battery comparment.</span>")
+			opened = FALSE
+			update_icon()
+			return
+	if(istype(W,/obj/item/weapon/stock_parts/cell))
+		if(opened)
+			if(!cell)
+				if(!user.transferItemToLoc(W, src))
+					return
+				to_chat(user, "<span class='notice'>You insert the [W] into \the [src].</span>")
+				cell = W
+				update_icon()
+				return
+			else
+				to_chat(user, "<span class='notice'>The [src] already has \the [cell] installed!</span>")
+				return
+
+	if(!user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to use \the [src]!</span>")
+		return
+
+	if(!cell)
+		to_chat(user, "<span class='warning'>The [src] doesn't have a power cell installed!</span>")
+		return
+
+	if(!cell.charge)
+		to_chat(user, "<span class='warning'>The [src]'s battery is dead!</span>")
+		return
+
+	var/obj/item/weapon/stock_parts/cell/C = W.get_cell()
+	if(C)
+		if(C.charge == C.maxcharge)
+			to_chat(user, "<span class='notice'>The [W] is fully charged!</span>")
+			return
+		user.visible_message("[user] starts recharging the [W] with \the [src]",\
+							"<span class='notice'>You start recharging the [W] with \the [src]</span>")
+		if(do_after(user, 60, target = W, progress = 1)) //Charging items like this is slower because I know someone will bite my head off otherwise.
+			induce(C)
+			user.visible_message("[user] recharged the [W]!",\
+								 "<span class='notice'>You recharged the [W]!</span>")
+			return
+
+	..()
+
+/obj/item/weapon/inducer/attack(mob/M, mob/user)
+	if(!user.IsAdvancedToolUser())
+		to_chat(user, "<span class='warning'>You don't have the dexterity to use \the [src]!</span>")
+		return
+
+	if(!cell)
+		to_chat(user, "<span class='warning'>The [src] doesn't have a power cell installed!</span>")
+		return
+
+	if(!cell.charge)
+		to_chat(user, "<span class='warning'>The [src]'s battery is dead!</span>")
+		return
+
+	var/obj/item/weapon/stock_parts/cell/C = M.get_cell()
+	if(C)
+		if(C.charge == C.maxcharge)
+			to_chat(user, "<span class='notice'>[M] is fully charged!</span>")
+			return
+
+		user.visible_message("[user] starts recharging [M] with \the [src]",\
+							"<span class='notice'>You start recharging [M] with \the [src]</span>")
+		if(do_after(user, 10, target = M, progress = 1)) //Charging borgs is fast, since they can have high capacity cells inside.
+			induce(C)
+			user.visible_message("[user] recharged [M]!",\
+								 "<span class='notice'>You recharged [M]!</span>")
+		return
+	..()
+
+
+/obj/item/weapon/inducer/attack_hand(mob/user)
```

Comment:
Use `attack_self()` here. You may then delete line 141 entirely, as all of these checks are not needed.